sdk/storage/inc/datalake/directory_client.hpp

```diff
@@ -0,0 +1,157 @@
+     */
+    explicit DirectoryClient(
+        const std::string& directoryUri,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        const DirectoryClientOptions& options = DirectoryClientOptions());
+
+    /**
+     * @brief Anonymous/SAS/customized pipeline auth.
+     * @param directoryUri The URI of the file system this client's request targets.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit DirectoryClient(
+        const std::string& directoryUri,
+        const DirectoryClientOptions& options = DirectoryClientOptions());
+
+    /**
+     * @brief Gets the directory's primary uri endpoint. This is the endpoint used for blob
+     * service interop.
+     *
+     * @return The directory's primary uri endpoint.
+     */
+    std::string GetUri() const { return m_blobClient.GetUri(); }
+
+    /**
+     * @brief Gets the directory's primary uri endpoint. This is the endpoint used for dfs
+     * endpoint only operations
+     *
+     * @return The directory's primary uri endpoint.
+     */
+    std::string GetDfsUri() const { return m_dfsUri.ToString(); }
+
+    // Feature not yet enabled for 2019-12-12.
+    ///**
+    // * @brief Sets POSIX access control rights on files and directories under given directory
+    // *        recursively.
+    // * @param mode Mode PathSetAccessControlRecursiveMode::Set sets POSIX access control rights on
+    // *             files and directories, PathSetAccessControlRecursiveMode::Modify modifies one
+    // or
+    // *             more POSIX access control rights  that pre-exist on files and directories,
+    // *             PathSetAccessControlRecursiveMode::Remove removes one or more POSIX access
+    // *             control rights that were present earlier on files and directories
+    // * @param acls Sets POSIX access control rights on files and directories. Each access control
+    // *             entry (ACE) consists of a scope, a type, a user or group identifier, and
+    // *             permissions.
+    // * @param options Optional parameters to set an access control recursively to the resource the
+    // *                directory points to.
+    // * @return Azure::Core::Response<DirectorySetAccessControlRecursiveInfo>
+    // * @remark This request is sent to dfs endpoint.
+    // */
+    // Azure::Core::Response<DirectorySetAccessControlRecursiveInfo> SetAccessControlRecursive(
+    //    PathSetAccessControlRecursiveMode mode,
+    //    std::vector<Acl> acls,
+    //    const SetAccessControlRecursiveOptions& options = SetAccessControlRecursiveOptions())
+    //    const;
+
+    /**
+     * @brief Create a directory. By default, the destination is overwritten and
+     *        if the destination already exists and has a lease the lease is broken.
+     * @param options Optional parameters to create the directory the path points to.
+     * @return Azure::Core::Response<DirectoryInfo>
```

Comment:
Simply putting return type here doesn't help at all.

Reply:
ditto for others

Reply:
I think there isn't anymore needed to be documented except for the return type. Descriptive words do not add any more clarity to the existing simple enough structures.

Reply:
No, only putting a return type here is not documentation.

Reply:
Added an issue to track this: https://github.com/Azure/azure-sdk-for-cpp/issues/331